### ISSUE TITLE
How to set GpuOptions in ML.NET

### ISSUE THREAD

**Issue by DirkKramer** [dotnet/machinelearning#5590]
System information
- Windows 10
- .NET Framework 4.8
- CUDA 10.1
- Cudnn 7.6.5
- Scisharp.Tensorflow.redist.window.gpu 2.3.1
- Microsoft.ML 1.5.4 (vision, imageanalytics)

Hardware information
-AMD Ryzen 7 4800
-Nvidia GTX 1650

Training and predictions fail on GPU because of probably Memory out of range exception.   Everything works fine on my other laptop and desktop. but this laptop gives errors (maybe because of AMD??) . 

![image](https://user-images.githubusercontent.com/46933603/104698269-637b0800-5711-11eb-8f9a-f9077d7472a5.png)

- **What did you do?**
-I tried different combinations of CUDA-----cuDNN----Scisharp tensorflow redist gpu. (same error)
-I tried rebooting.
-I read on different issues on tensorflow.net and stackoverflow that you can set the gpuoptions. but I dont know how to do that in ML.NET.

- **What do I expect?**
Correct training and predictions on GPU without exceptions.

`var mlContext = new MLContext(seed: 1);`
        
            // 2. Load the initial full image-set into an IDataView and shuffle so it'll be better balanced
            IEnumerable<ImageData> images = LoadImagesFromDirectory(folder: fullImagesetFolderPath, useFolderNameAsLabel: true);
            IDataView fullImagesDataset = mlContext.Data.LoadFromEnumerable(images);
            IDataView shuffledFullImageFilePathsDataset = mlContext.Data.ShuffleRows(fullImagesDataset);

            // 3. Load Images with in-memory type within the IDataView and Transform Labels to Keys (Categorical)
            IDataView shuffledFullImagesDataset = mlContext.Transforms.Conversion.
                    MapValueToKey(outputColumnName: "LabelAsKey", inputColumnName: "Label", keyOrdinality: KeyOrdinality.ByValue)
                .Append(mlContext.Transforms.LoadRawImageBytes(
                                                outputColumnName: "Image",
                                                imageFolder: fullImagesetFolderPath,
                                                inputColumnName: "ImagePath"))
                .Fit(shuffledFullImageFilePathsDataset)
                .Transform(shuffledFullImageFilePathsDataset);

            // 4. Split the data 80:20 into train and test sets, train and evaluate.
            var trainTestData = mlContext.Data.TrainTestSplit(shuffledFullImagesDataset, testFraction: 0.2);
            IDataView trainDataView = trainTestData.TrainSet;
            IDataView testDataView = trainTestData.TestSet;

            // 5. Define the model's training pipeline using DNN default values            
            var pipeline = mlContext.MulticlassClassification.Trainers
                    .ImageClassification(featureColumnName: "Image",
                                            labelColumnName: "LabelAsKey",
                                            validationSet: testDataView)
                .Append(mlContext.Transforms.Conversion.MapKeyToValue(outputColumnName: "PredictedLabel",
                                                                      inputColumnName: "PredictedLabel"));

            // 6. Train/create the ML model       
            ITransformer trainedModel = pipeline.Fit(trainDataView);`
